### DISCUSSION TITLE
Communication with FT2232D over USB

### DISCUSSION THREAD

**Discussion by S0UL4** [zephyrproject-rtos/zephyr#70643] [Q&A]
Hello guys,
I'm using EVK-ANNA-B402 with Zephyr and trying to find a way to communicate with an FT2232D over USB in a bidirectional way.
Anyone succeeded to do that? 

The FT2232D belongs to the Vendor-Specific USB class and cannot be used with the CDC-USB class. Therefore, I am exploring alternatives and the development of a driver for the FT2232D appears to be quite complex :c 

Here is the device to communicate with : 

![image](https://github.com/zephyrproject-rtos/zephyr/assets/62818840/cd8a71f2-98ab-4f11-8469-acd05a85b6cf)

Any assistance would be greatly appreciated ! 
Thank you in advance!

**Comment by ndrs-pst** [comment]
@S0UL4 How about https://github.com/hathach/tinyusb? It seems to support FTDI.
You may try to port it to Zephyr as a module or just try it with another OS abstraction layer to see if it fits your requirements.
